geth/peers/peerpool.go

```diff
@@ -62,13 +70,11 @@ type peerInfo struct {
 
 // PeerPool manages discovered peers and connects them to p2p server
 type PeerPool struct {
+	opts *Options
+
 	// config can be set only once per pool life cycle
-	config            map[discv5.Topic]params.Limits
-	fastSync          time.Duration
-	slowSync          time.Duration
-	discServerTimeout time.Duration
-	cache             *Cache
-	stopOnMax         bool
+	config map[discv5.Topic]params.Limits
```

Comment:
does it make sense to move topic=limits to options?

Reply:
I was thinking about `Options` fields as optional or having a default value. `config` seems like a required argument in a real-world scenario.